### ISSUE TITLE
Move irtkVTK.h from Image to Common

### ISSUE THREAD

**Issue by schuhschuh** [BioMedIA/irtk-refactored-deprecated#2]
In my personal development branch, I moved irtkVTK.h from the Image module to Common and renamed it to irtkVtk.h (to follow naming convention of other similar files, such as irtkUmfpack.h). This issue is to remind of merging this change from my branch into master.

**Comment by ghisvail** [comment]
Any specific rationales for this change ?

**Comment by schuhschuh** [comment]
The file is not specific to the Image module. The Common module should provide other modules with the tools to take advantage of common external libraries.

**Comment by ghisvail** [comment]
Fair point. That means Common will have optional linkage to VTK.

**Comment by schuhschuh** [comment]
Yes, the entire IRTK project always had optional linkage to VTK. There are some "core" applications that used it independently of the Image module. Maybe these can be made part of the PolyData module instead and every module that uses VTK needs to depend on this module... ? Not something I had in mind yet as the PolyData module mainly served the purpose of dealing with surface meshes (but is no longer limited to it, but point sets in general).

**Comment by ghisvail** [comment]
I can't comment on the feasibility of completely encapsulating our usage of VTK. It would be a nice goal towards having IRTK as a proper library.

**Comment by schuhschuh** [comment]
I would say that mostly depends on the effort you want to put into wrapping the essential VTK data object types such as vtkPointSet, vtkPolyData, vtkImageData, vtkDataArray, (vtkPoints -> irtkPointSet) ... other than that, all uses of the VTK algorithms is (or can be) hidden within the IRTK classes. Just the data containers need to be exchanged.

In my PolyData module implementations, I have for sake of simplicity used the VTK data types directly as class attributes and function arguments. Hence, all this would need to be replaced by IRTK proxy types.
